src/js/techmix_sector.js

```diff
@@ -0,0 +1,381 @@
+			.attr('preserveAspectRatio', 'xMinYMin meet')
+			.attr('style', 'max-width: 100%; height: auto;');
+
+		// Filter the data
+		let sector = document.querySelector('#sector_selector').value,
+			asset_class = document.querySelector('#asset_class_selector').value,
+			scenario_source = document.querySelector('#scenario_source_selector').value,
+			scenario = document.querySelector('#scenario_selector').value,
+			equity_market = document.querySelector('#equity_market_selector').value,
+			benchmark = document.querySelector('#benchmark_selector').value;
+		let subdata = data
+			.filter((d) => d.asset_class == asset_class)
+			.filter((d) => d.ald_sector == sector)
+			.filter((d) => d.scenario_source == scenario_source)
+			.filter((d) => d.scenario == scenario)
+			.filter((d) => d.equity_market == equity_market)
+			.filter((d) => d.this_portfolio | (d.portfolio_name == benchmark));
+
+		if (subdata.length == 0) {
+			container_div.querySelector('svg').innerHTML = '';
+			return;
+		}
+
+		// create the stacked data for plotting
+		let uniqueYears = d3.map(subdata, (d) => d.year).keys();
+		let uniqueValueTypes = [];
+		uniqueYears.forEach((item, idx) => {
+			uniqueValueTypes[idx] = d3
+				.map(
+					subdata.filter((d) => d.year == item),
+					(d) => d.val_type
+				)
+				.keys();
+		});
+		let subdataTechPerYear = [];
+		uniqueYears.forEach((item, idx) => {
+			subdataTechPerYear[idx] = {};
+			subdataTechPerYear[idx]['year'] = item;
+			subdataTechPerYear[idx]['stackedData'] = getStackedDataPerYear(
+				subdata,
+				uniqueValueTypes[idx],
+				item
+			);
+			subdataTechPerYear[idx]['greenBars'] = getGreenBarsValsPerYear(
+				subdata,
+				uniqueValueTypes[idx],
+				item
+			);
+		});
+		// subgroups contains all possible technologies for sector
+		let subgroups0 = new Set(d3.map(subdataTechPerYear[0].stackedData, (d) => d.key).keys());
+		let subgroups1 = new Set(d3.map(subdataTechPerYear[1].stackedData, (d) => d.key).keys()) 
+
+		let subgroups = Array.from(subgroups0.union(subgroups1));
+
+		// Declare the x scale
+		const x = d3
+			.scaleLinear()
+			.domain([0, 1])
+			.range([marginLeft, width - marginRight]);
+
+		// Declare the y scales
+		const y0 = d3
+			.scaleBand()
+			.domain(uniqueYears)
+			.rangeRound([marginTop, height - marginBottom])
+			.paddingInner(0.1);
+
+		const yCurrent = d3
+			.scaleBand()
+			.domain(uniqueValueTypes[0])
+			.rangeRound([0, y0.bandwidth()])
+			.paddingInner(0.1);
+
+		const yFuture = d3
+			.scaleBand()
+			.domain(uniqueValueTypes[1])
+			.rangeRound([0, y0.bandwidth()])
+			.paddingInner(0.1);
+
+		let barHeight = Math.min(
+			yCurrent.bandwidth() - yCurrent.bandwidth() / 5,
+			yFuture.bandwidth() - yFuture.bandwidth() / 5
+		);
+
+		// Declare the colours
+		const color = d3
+			.scaleOrdinal()
+			.domain(subgroups)
+			.range(d3.schemeSpectral[subgroups.length])
+			.unknown('#ccc');
+
+		// Add rectangles for each stacked bar - TODO: rewrite into two
```

Comment:
NIT: @MonikaFu in general, we should aim to NOT have any TODOs on `main`. So my suggestion would be to please review all remaining TODOs in the comments and either:
* handle them (if directly relevant to this particular PR)
OR
* open a new GH issue explaining what should be done